sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1485,6 +1485,13 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
     .stringConf
     .createWithDefault(false.toString)
 
+  val FOLDABLE_NON_LIT_ALLOWED = conf("spark.rapids.sql.test.isFoldableNonLitAllowed")
```

Comment:
I'm not sure we need this at all.  The foldable check was put in place a long time ago when each expression had to have explicit support for scalar values. We refectored the code a while ago and removed that. Just the fact that it works for testing means that the refactor was correct and I think we can just delete the check entirely.

Reply:
should I do it in this PR or in another separate one?

Reply:
I am fine with doing it later as this is an internal config, and it might be a large-ish change.  But at the same time I don't want to forget about it.

Reply:
this issue will be tracked in https://github.com/NVIDIA/spark-rapids/issues/10887